src/world/GameWorldView.cpp

```diff
@@ -290,7 +290,7 @@ void GameWorldView::DrawGUI(const RoadBuildState& rb, const TerrainRenderer& ter
                     continue;
 
                 if((gwv.IsRoadAvailable(rb.mode == RM_BOAT, curPt) && gwv.IsOwner(curPt) && GetWorld().IsPlayerTerritory(curPt))
-                   || (gwv.GetBQ(curPt) == BQ_FLAG))
+                   || (gwv.GetBQ(curPt) == BQ_FLAG) || (GetWorld().GetNO(curPt)->GetType() == NOP_FLAG && curPt != rb.start))
```

Comment:
Can the last condition become a problem if the flag belongs to an other player? Maybe the last 2 conditions (Owner and player territory) should always apply. But would need to think carefully about this.